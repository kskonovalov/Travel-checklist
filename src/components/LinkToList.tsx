import React, { useState, useRef } from 'react';
import { InputAdornment, TextField } from '@material-ui/core';
import { Link as LinkIcon } from '@material-ui/icons';

const LinkToList = () => {
  const [helperText, setHelperText] = useState<string>(
    'Нажмите на скрепку, чтобы скопировать'
  );

  const textInputRef = useRef(null);
  return (
    <TextField
      id="standard-basic"
      label="Ссылка на Ваш личный чеклист путешественника"
      fullWidth={true}
      value={`${window.location.href}`}
      helperText={helperText}
      inputRef={textInputRef}
      InputProps={{
        startAdornment: (
          <InputAdornment position="start">
            <LinkIcon
              style={{ cursor: 'pointer' }}
              onClick={() => {
                console.log(textInputRef);
                // @ts-ignore
                textInputRef.current.select();
                document.execCommand('copy');
                setHelperText('Скопировано в буфер обмена!');
              }}
            />
          </InputAdornment>
        )
      }}
    />
  );
};

export default LinkToList;
