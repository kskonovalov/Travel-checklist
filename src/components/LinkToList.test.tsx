import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import LinkToList from './LinkToList';

configure({ adapter: new Adapter() });

const setUp = ({ text }: { text: string }) => {
  return <LinkToList />;
};

describe('Component: ErrorMessage', () => {
  test('Should match snapshot', () => {
    const text = 'Test text';
    const component = shallow(setUp({ text }));
    expect(component).toMatchSnapshot();
  });
});
