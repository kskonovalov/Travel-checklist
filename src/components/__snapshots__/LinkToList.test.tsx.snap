// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ErrorMessage Should match snapshot 1`] = `
<WithStyles(ForwardRef(TextField))
  InputProps={
    Object {
      "startAdornment": <ForwardRef(WithStyles)
        position="start"
      >
        <UNDEFINED
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        />
      </ForwardRef(WithStyles)>,
    }
  }
  fullWidth={true}
  helperText="Нажмите на скрепку, чтобы скопировать ссылку на Ваш лист"
  id="standard-basic"
  inputRef={
    Object {
      "current": null,
    }
  }
  label="Ссылка на Ваш личный чеклист путешественника"
  value="http://localhost/"
/>
`;
